docs/proposals/PROPOSAL_E.md

```diff
@@ -157,6 +163,51 @@ If a query results in no referrers found, an empty manifest list MUST be returne
 - The registry SHOULD return fewer than `n` results when the generated Index would exceed the recommended maximum manifest size.
 - The `Link` HTTP header is included in the response when additional results are available and is set to the URL for the next page of results.
 
+#### Filtering
+A limited ability to filter results based on `artifactType` will enable the referrers API to scale as the number of artifact use cases grows.
+ - The registry SHOULD allow for filtering based on `artifactType`.
+ - Clients MAY request results be filterd by adding the `artifactType` query parameter to requests to the referrers API.
+ - The value for `artifactType` MUST be the IANA media type of the artifact to be returned.
+
+Example request with filtering:
+ ```text
+GET /v2/<name>/referrers/<ref>?artifactType=application/vnd.example.icecream.v1
+```
+
+Response:
+```jsonc
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "artifactType": "application/vnd.example.icecream.v1", // pulled up from manifest
+      "size": 1234,
+      "digest": "sha256:a1a1a1...",
+      "annotations": [
+        // annotations pulled up from manifest
+        "org.opencontainers.artifact.created": "2022-01-01T14:42:55Z",
+        "org.example.icecream.flavor": "chocolate"
+      ]
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "artifactType": "application/vnd.example.icecream.v1",
+      "size": 1234,
+      "digest": "sha256:a2a2a2...",
+      "annotations": [
+        "org.opencontainers.artifact.created": "2022-01-01T15:24:30Z",
+        "org.example.icecream.flavor": "vanilla"
+      ]
+    }
+  ],
+  "annotations": {
+    "artifactType": "application/vnd.example.icecream.v1"
+  }
```

Comment:
should this be codified into the proposal?

Reply:
So from the discussion just an indication of the field without he values makes the response for unmatched artifacts cachable.